src/compiler/checker.ts

```diff
@@ -21205,7 +21205,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const id = getRelationKey(source, target, intersectionState, relation, /*ignoreConstraints*/ false);
             const entry = relation.get(id);
             if (entry !== undefined) {
-                if (reportErrors && entry & RelationComparisonResult.Failed && !(entry & RelationComparisonResult.Reported)) {

```

Comment:
if this fix turns out to be in a good direction then `RelationComparisonResult.Reported` should be cleaned up further